src/internal/operations/manifests_test.go

```diff
@@ -0,0 +1,612 @@
+		reasons       []kopia.Reason
+		getMeta       bool
+		assertErr     assert.ErrorAssertionFunc
+		assertB       assert.BoolAssertionFunc
+		expectDCS     []data.Collection
+		expectNilMans bool
+	}{
+		{
+			name: "don't get metadata, no mans",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "don't get metadata",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "", "")},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "don't get metadata, incomplete manifest",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "ir", "")},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "fetch manifests errors",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mrErr:        assert.AnError,
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.Error,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "verify distinct bases fails",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans: []*kopia.ManifestEntry{
+					makeMan(path.EmailCategory, "", ""),
+					makeMan(path.EmailCategory, "", ""),
+				},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError, // No error, even though verify failed.
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "no manifests",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError,
+			assertB:   assert.True,
+			expectDCS: nil,
+		},
+		{
+			name: "only incomplete manifests",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans: []*kopia.ManifestEntry{
+					makeMan(path.EmailCategory, "ir", ""),
+					makeMan(path.ContactsCategory, "ir", ""),
+				},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError,
+			assertB:   assert.True,
+			expectDCS: nil,
+		},
+		{
+			name: "man missing backup id",
```

Comment:
Could you please add a test for missing details ID as well please?

Reply:
absolutely